### PR TITLE
Initial stab at Chromium compatibility

### PR DESCRIPTION
**Pull request by lidel**
This PR provides initial support for Chromium 58+ browsers (https://github.com/ipfs/in-web-browsers/issues/54, #218).

### Running in Chromium

First, build it manually:

 ```bash
npm install
npm run build
 ```
Then open up `chrome://extensions` in Chromium-based browser, enable "Developer mode", click "Load unpacked extension..." and point it at `add-on/manifest.json`.

It [should load just fine](https://blog.codinghorror.com/content/images/uploads/2007/03/6a0120a85dcdae970b0128776ff992970c-pi.png), with some minor warnings (see _known issues_ below):

>  2017-05-13-211205_897x506_scrot

### Brief list of issues fixed in this PR

- Chromium does not have `browser` object and requires a polyfill for standardized APIs (which is a subset of APIs provided by Firefox): https://github.com/mozilla/webextension-polyfill 
- Only one of `browser_action`, `page_action` can be specified in Chromium 
- Chromium has some additional restrictions regarding locale files which needs to be addressed (fixed by removing `-` from key names)
- Chromium complains about some i18 placeholders  present in local files 
- Chromium does not support "alpha/beta" suffix in `version` in manifest 
- Chromium does not accept SVG as browserAction icon, PNG needs to be provided instead (https://stackoverflow.com/q/38457208, https://bugs.chromium.org/p/chromium/issues/detail?id=647182, https://bugs.chromium.org/p/chromium/issues/detail?id=29683) 
- Chromium sends `Origin: null` with XHR requests for `/api/v0/swarm/peers` which causes `403 - Forbidden` error
- Chromium does not ship with native CSS classes that are shipped with Firefox 
- Right-click "Upload to IPFS" does not work due to [Origin mismatch](https://cloud.githubusercontent.com/assets/157609/26028017/7e694404-3818-11e7-90c8-3eee78b0bbb8.png)

### Known Issues

Remaining Issues that are not covered by this PR:
- Chromium does not accept Alarm periods less than minimum of 1 minutes. This means "ipfs-api-status-update" will fire approximately every 1 minutes (instead of every 3 seconds). We can leave it that way, or rewrite API polling logic to use something else than Alarms API.
- No support for `web+foo` protocols. Chromium does not support `protocol_handlers` in manifest, this needs to be detected and manual handler needs to be installed somehow.

### CODE REVIEW

**Comment by lidel** [comment]
Are we good to merge this? If so, hit me up with review approval.
I'd like to release it into Firefox's Development Channel as `v2.0.0alpha5`

**Comment by whyrusleeping** [comment]
loading it into chrome i get:
```
Failed to load extension from: ~/gopkg/src/github.com/ipfs/ipfs-companion/add-on
Invalid placeholder errorMsg for key notify_inlineErrorMsg
```
Followed by the entire manifest.json in a little scrollybox

**Comment by whyrusleeping** [comment]
chromium version: 58.0.3029.110

**Comment by lidel** [comment]
@whyrusleeping  Are you sure you loaded code from `chromium-compat` branch? 
Mentioned bug was fixed in https://github.com/ipfs/ipfs-companion/pull/250/commits/6f007412093033837fddc1648bca086ee9f5c65d

**Comment by whyrusleeping** [comment]
@lidel youre right, i missed the checkout a branch bit. Works! :+1: :+1:

File: add-on/src/lib/common.js [hunk 1]
```diff
@@ -383,17 +414,60 @@ function updateBrowserActionBadge () {
...
+  let canvas = document.createElement('canvas')
+  let context = canvas.getContext('2d')
+  context.clearRect(0, 0, size, size)
+  context.drawImage(icon, 0, 0, size, size)
+  return context.getImageData(0, 0, size, size)
+}
+*/
+
 // OPTIONS
 // ===================================================================
```
Comment (diasdavid) [inline add-on/src/lib/common.js]: Wouldn't it make more sense to modularize these sections?

**Comment by diasdavid** [comment]
Ah, only see that purple merged label after. awesome!

File: add-on/src/lib/common.js (hunk 1 above)
Comment (lidel) [inline add-on/src/lib/common.js]: Yes, at some point. 
I've added those "section" comments, as `common.js` grew organically.  

I was not sure what is the best course of action for "webextension modularization", as Firefox WX ecosystem was still brand new when I started the rewrite from legacy SDK.

In Chrome world people use [ES6 modules](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import), but feature is not implemented in any browsers natively at this time and requires you to transpile everything to a single `.js` file ([example](https://github.com/samuelsimoes/chrome-extension-webpack-boilerplate)).

I avoided it and right now add-on code runs "as-is" (no transpilling), but I am not sure if this approach can be maintained in the long run, especially now that we support multiple browser vendors.

If you feel it is the time, create a new Issue and lcopy/link this comment there.
Alternatively, we leave it be a monolith until ES6 modules are natively supported by both Firefox and Chrome.

**Pull request merged**